contrib/babelfishpg_common/src/collation.c

```diff
@@ -601,6 +629,33 @@ translate_collation(const char *collname, bool check_for_server_collation_name_g
 	return idx;
 }
 
+/* Translate BBF collation to it's closest SQL Server Collation. */
+char *
+translate_collation_bbf(const char *collname)
+{
+	char *res = NULL;
+	int first = 0;
+	int last = TOTAL_COLL_TRANSLATION_COUNT_TSQL - 1;
+	int middle = 16; /* optimization: usually it's the default collation. */
+	int compare;
+
+	while (first <= last)
+	{
+		compare = pg_strcasecmp(coll_translations_tsql[middle].from_collname, collname);
+		if (compare < 0)
+			first = middle + 1;
+		else if (compare == 0)
+		{
+			return (char*)(coll_translations_tsql[middle].to_collname);
+		}
+		else
+			last = middle - 1;
+
+		middle = (first + last) / 2;
+	}
+	return res;
```

Comment:
we don't need res anywhere except this place, let's return NULL directly from here.